src/stream.lisp

```diff
@@ -92,3 +147,67 @@
   `(let ((,var (make-string-output-stream)))
      ,@body
      (get-output-stream-string ,var)))
+
+;;; replace sequences - http://clhs.lisp.se/Body/f_replac.htm
+;;;   string <- string
+;;;   vector <- vector
+;;;   otherwise =>error
+
+(defun %replace-seq-eql (x y)
+  (typecase x
+    (number (and (numberp y) (= x y)))
+    (character (and (characterp y) (char-equal x y)))
+    (string
+     (and (stringp y)
+          (equal x y) ))
+    (vector
+     (and (vectorp y)
+          (let ((lex (length x)))
+            (= lex (length y))
+            (dotimes (i lex t)
+              (when (not (%replace-seq-eql (aref x i) (aref y i)))
+                (return nil))))))
+    (t (equal x y))))
+
+(defun %replace-seq (seq-1 seq-2 start1 end1 start2 end2)
+  (let* ((trimed-end (min (- end1 start1) (- end2 start2)))
+         (back nil))
+    (setq end1 (+ start1 trimed-end)
+          end2 (+ start2 trimed-end))
+    ;; set copy bacward flag
+    (when (and (%replace-seq-eql seq-1 seq-2)
+               (<= start2 start1)
+               (or (and (<= start1 start2) (< start2 end1))
+                   (and (< start1 end2) (<= end2 end1))
+                   (and (<= start2 start1) (< start1 end2))
+                   (and (< start2 end1) (<= end1 end2))))
+      (when (eq start1 start2)
+        ;; nothing to copy
+        (return-from %replace-seq seq-1))
+      (setq back t))
+    (cond (back
+           (dotimes (i trimed-end seq-1)
+             (setf (aref seq-1 (- (+ start1 trimed-end) i 1))
+                   (aref seq-2 (- (+ start2 trimed-end) i 1)))))
+          (t
+           (dotimes (i trimed-end seq-1)
+             (setf (aref seq-1 (+ start1 i))
+                   (aref seq-2 (+ start2 i))))))
+    ))
+
+(defun replace (sequence-1
```

Comment:
I think `replace` should be in sequence.lisp.

Reply:
I also thought that this function is in sequence.lisp.
But one day she was needed.

Reply:
>Why the change from (format t to (format *standard-output? Shouldn't it be the same?

Yes! :-)

```lisp
1. *standard-output* eq t
2. t eq *standard-output*
3. see 1
```

Reply:
>I also thought that this function is in sequence.lisp.
Everything is simple.
The code uses macro ASSERT, which is defined in CONDITION.lisp

See comment for commit:
https://github.com/jscl-project/jscl/pull/446/commits/2235edd7f05ddc06e58a306cc9340b5d300de071

It would be necessary to moved macro ASSERT to BOOT.lisp, but I'm not ready to do such patches now.

Reply:
Doesn't it work if we just move it there?  The macro itself it pretty simple and it just compiles to a function call.  Note that you don't need to move `%%assert` itself,  just the defmacro.

I think that should work 🙏

Reply:
No problem, boss ;)
Let's do it nowю

Reply:
done